hathor/cli/util.py

```diff
@@ -164,17 +185,37 @@ def kwargs_formatter(_, __, event_dict):
             structlog.processors.StackInfoRenderer(),
             structlog.processors.format_exc_info,
             structlog.stdlib.ProcessorFormatter.wrap_for_formatter,
-            # structlog.twisted.JSONRenderer(),
         ],
-        context_class=dict,
+        context_class=OrderedDict,
         logger_factory=structlog.stdlib.LoggerFactory(),
         wrapper_class=structlog.stdlib.BoundLogger,
         cache_logger_on_first_use=True,
     )
 
-    twisted.python.log.startLoggingWithObserver(twisted.logger.STDLibLogObserver(), setStdout=capture_stdout)
+    twisted_logger = structlog.get_logger('twisted')
+
+    def twisted_structlog_observer(event):
+        try:
+            level = twisted_to_logging_level.get(event.get('log_level'), logging.INFO)
+            kwargs = {}
+            msg = ''
+            if not msg and 'log_format' in event:
+                msg = event['log_format'].format(**event)
+            if not msg and 'format' in event:
+                msg = event['format'] % event
+            failure = event.get('log_failure')
+            if failure is not None:
+                kwargs['exc_info'] = (failure.type, failure.value, failure.getTracebackObject())
+            twisted_logger.log(level, msg, **kwargs)
+        except Exception as e:
+            print('error when logging event', e)
```

Comment:
Should we exit the process?

Reply:
I'd rather not. I have on idea what could cause this. And if it turn out to be "frequent" we're going to have a bad time managing our nodes that don't restart automatically and even if they did it would take a good while to load/sync and be up and running again.

Let's first see how it goes.